### PR TITLE
version-2 part 4

### PR DESCRIPTION
**Pull request by shinokada**
### 🔗 Related issue (optional)

Closes #

---

## 📑 Description

---

## 🔍 PR Type

- [ ] Bug fix
- [ ] Feature
- [ ] Documentation
- [ ] Refactor / Code cleanup
- [ ] Build / Tooling
- [ ] Other (please describe)

---

## 🚦 PR Status

- [ ] Draft (work in progress, not ready for review)
- [ ] Ready for review ✅

---

## ✅ Checklist

- [ ] My code follows the existing code style
- [ ] I have run `pnpm lint && pnpm check && pnpm test:e2e` and all tests pass
- [ ] CoderabbitAI review has been completed and **actionable suggestions were reviewed**
- [ ] I have updated documentation if my changes require it
- [ ] My PR is based on the latest `main` branch (not the published npm version)
- [ ] I have checked accessibility where applicable (ARIA, keyboard nav, etc.)
- [ ] I have reviewed the rendered component in the browser

---

## 🧪 Screenshots / Demos (optional)

---

## ⚠️ Breaking Changes (optional)

---

## Release Notes

* **New Features**
  * Added Theme Selector component with support for multiple built-in themes
  * Implemented accessibility support for reduced-motion preferences across components
  * Added theme configuration and dynamic theme switching capabilities

* **Enhancements**
  * Made drawer dismissibility behavior configurable
  * Updated component styling for improved visual consistency

* **Documentation**
  * Added Theme Selector documentation and usage guide
  * Added test coverage documentation for new features

* **Tests**
  * Added comprehensive test suites for Theme Selector functionality

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/routes/utils/ThemeSelector.svelte [hunk 1]
```diff
@@ -24,12 +23,14 @@
...
-  class="gap-2 rounded-xl border-0 px-1.5 py-1.5 shadow-none focus:ring-gray-100 dark:focus:ring-gray-700"
+  class="gap-2 rounded-sm border-0 px-1 py-1 shadow-none focus:ring-gray-100 dark:focus:ring-gray-700"
   aria-haspopup="true"
   aria-expanded={isOpen}
   aria-label="Select Theme"
 >
-  <PaletteOutline class="h-6 w-6" />
+  <svg class="w-4.5 h-4.5" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none" viewBox="0 0 24 24">
+    <path stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 7h.01m3.486 1.513h.01m-6.978 0h.01M6.99 12H7m9 4h2.706a1.957 1.957 0 0 0 1.883-1.325A9 9 0 1 0 3.043 12.89 9.1 9.1 0 0 0 8.2 20.1a8.62 8.62 0 0 0 3.769.9 2.013 2.013 0 0 0 2.03-2v-.857A2.036 2.036 0 0 1 16 16Z"></path>
+  </svg>
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/routes/utils/ThemeSelector.svelte]: It is calculated by width: calc(var(--spacing) * 4.5);.

File: src/tests/theme-selector/themeStore.test.ts [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+      expect(selectedTheme).toBeDefined();
+      expect(selectedTheme?.id).toBe("enterprise");
+      expect(selectedTheme?.name).toBe("Enterprise");
+    });
+
+    test("returns undefined for invalid theme", () => {
+      // This shouldn't happen in practice, but testing edge case
+      const selectedTheme = getSelectedTheme();
+      expect(selectedTheme).toBeDefined();
+    });
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/theme-selector/themeStore.test.ts]: fixed.

File: src/tests/theme-selector/themeStore.test.ts [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+      // Trigger onerror
+      const fontLink = mockLinks.find((link) => link.id === "dynamic-theme-font");
+      if (fontLink?.onerror && typeof fontLink.onerror === "function") {
+        fontLink.onerror(new Event("error"), "", 0, 0, undefined);
+      }
+
+      expect(consoleWarnSpy).toHaveBeenCalled();
+
+      consoleWarnSpy.mockRestore();
+    });
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/theme-selector/themeStore.test.ts:287]: fixed.

File: src/tests/theme-selector/themeStore.test.ts [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+    test("logs error for theme CSS loading failure", () => {
+      const consoleErrorSpy = vi.spyOn(console, "error").mockImplementation(() => {});
+
+      loadTheme("default");
+
+      // Trigger onerror
+      const themeLink = mockLinks.find((link) => link.id === "dynamic-theme-css");
+      if (themeLink?.onerror && typeof themeLink.onerror === "function") {
+        themeLink.onerror(new Event("error"), "", 0, 0, undefined);
+      }
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/theme-selector/themeStore.test.ts:298]: fixed.

File: src/routes/docs/pages/theme-selector.md [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+
+```
+static/
+  └── themes/
+      ├── default-runtime.css
+      ├── minimal-runtime.css
+      ├── enterprise-runtime.css
+      ├── playful-runtime.css
+      └── mono-runtime.css
+```
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/routes/docs/pages/theme-selector.md]: fixed

File: src/lib/accordion/AccordionItem.svelte [hunk 6]
```diff
@@ -25,6 +26,14 @@
   // Check if transitionType is explicitly set to undefined in props
   const useTransition = $derived(transitionType === "none" ? false : ctxTransitionType === "none" ? false : true);
 
+  // Get respectReducedMotion from context (defaults to true)
+  const ctxRespectReducedMotion = $derived(ctx?.respectReducedMotion ?? true);
+
+  // Merge transition params with reduced motion handling
+  // Only check prefersReducedMotion in the browser to avoid SSR issues
+  const isBrowser = typeof window !== "undefined";
+  const effectiveTransitionParams = $derived(isBrowser && ctxRespectReducedMotion && prefersReducedMotion.current ? { duration: 0, ...transitionParams } : transitionParams);
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/accordion/AccordionItem.svelte]: fixed.

File: src/lib/alert/Alert.svelte [hunk 7]
```diff
@@ -41,6 +42,12 @@
     })
   );
 
+  // Check if running in browser to avoid SSR issues
+  const isBrowser = typeof window !== "undefined";
+
+  // Respect reduced motion preference by setting duration to 0
+  const effectiveParams = $derived(isBrowser && prefersReducedMotion.current ? { duration: 0, ...params } : params);
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/alert/Alert.svelte]: fixed.

File: src/lib/badge/Badge.svelte [hunk 8]
```diff
@@ -33,6 +34,12 @@
 
   const { base, linkClass } = $derived(badge({ color, size: large ? "large" : "small", rounded, border, href: !!href }));
 
+  // Check if running in browser to avoid SSR issues
+  const isBrowser = typeof window !== "undefined";
+
+  // Respect reduced motion preference by setting duration to 0
+  const effectiveParams = $derived(isBrowser && prefersReducedMotion.current ? { duration: 0, ...params } : params);
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/badge/Badge.svelte]: fixed.

File: src/lib/banner/Banner.svelte [hunk 9]
```diff
@@ -30,6 +31,12 @@
 
   const { base, insideDiv, dismissable: dismissableClass } = $derived(banner({ type, color }));
 
+  // Check if running in browser to avoid SSR issues
+  const isBrowser = typeof window !== "undefined";
+
+  // Respect reduced motion preference by setting duration to 0
+  const effectiveParams = $derived(isBrowser && prefersReducedMotion.current ? { duration: 0, ...params } : params);
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/banner/Banner.svelte]: fixed.

File: src/lib/dialog/Dialog.svelte [hunk 10]
```diff
@@ -29,7 +30,13 @@
...
   }: DialogProps = $props();
 
-  const paramsOptions = $derived(transitionParams ?? { duration: 100, easing: sineIn });
+  // Check if running in browser to avoid SSR issues
+  const isBrowser = typeof window !== "undefined";
+
+  // Respect reduced motion preference by setting duration to 0
+  const effectiveParams = $derived(
+    isBrowser && prefersReducedMotion.current ? { duration: 0, ...(transitionParams ?? { duration: 100, easing: sineIn }) } : (transitionParams ?? { duration: 100, easing: sineIn })
+  );
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/dialog/Dialog.svelte:39]: fixed.

File: src/lib/toast/Toast.svelte [hunk 11]
```diff
@@ -28,6 +29,12 @@
 
   const { base, icon: iconVariants, content, close } = $derived(toast({ color, position, align }));
 
+  // Check if running in browser to avoid SSR issues
+  const isBrowser = typeof window !== "undefined";
+
+  // Respect reduced motion preference by setting duration to 0
+  const effectiveParams = $derived(isBrowser && prefersReducedMotion.current ? { duration: 0, ...params } : params);
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/toast/Toast.svelte]: fixed.

File: src/lib/utils/Popper.svelte [hunk 12]
```diff
@@ -5,6 +5,7 @@
   import clsx from "clsx";
   import { sineIn } from "svelte/easing";
   import { fade } from "svelte/transition";
+  import { prefersReducedMotion } from "svelte/motion";
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/utils/Popper.svelte:8]: fixed.

File: src/lib/utils/Popper.svelte [hunk 13]
```diff
@@ -71,7 +72,14 @@
   });
 
   const paramsDefault = { duration: 100, easing: sineIn };
-  const paramsOptions = $derived(transitionParams ?? paramsDefault);
+
+  // Check if running in browser to avoid SSR issues
+  const isBrowser = typeof window !== "undefined";
+
+  // Respect reduced motion preference by setting duration to 0
+  const effectiveParams = $derived(isBrowser && prefersReducedMotion.current ? { duration: 0, ...(transitionParams ?? paramsDefault) } : (transitionParams ?? paramsDefault));
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/utils/Popper.svelte]: fixed.
